src/Components/List/List.jsx

```diff
@@ -37,21 +37,59 @@ export default function List({ items, token }) {
     history.push('/add-item');
   };
 
+  const [searchItem, setSearchItem] = React.useState('');
+  const handleChange = (e) => {
+    setSearchItem(e.target.value);
+  };
+
+  const resetSearch = () => {
+    setSearchItem('');
+  };
+
+  const results = !searchItem
+    ? items
+    : items.filter((item) =>
+        item.name.toLowerCase().includes(searchItem.toLocaleLowerCase()),
+      );
+
   return (
     <div className="List">
       {items.length === 0 ? (
         <section className="listContainer emptyList">
           <h3>
             Your shopping list is empty. Add a new item to start your list.
           </h3>
-          <button onClick={redirectPath}>Add New Item</button>
+          <button className="emptyButton" onClick={redirectPath}>
+            Add New Item
+          </button>
         </section>
       ) : (
-        <section className="listContainer">
-          <h3>Item List:</h3>
+        <section className="listContainer populatedList">
+          <h3>Smart Shopping List</h3>
+
+          <div className="search">
+            <label htmlFor="itemSearch">Search Items: </label>
+            <input
+              id="itemSearch"
+              type="text"
+              placeholder="Enter item name..."
+              value={searchItem}
+              onChange={handleChange}
+              aria-controls="itemsList"
+            />
+            {searchItem !== '' && (
+              <button
+                aria-label="Reset"
+                className="resetButton"
+                onClick={resetSearch}
+              >
+                x
+              </button>
+            )}
+          </div>
 
-          <ul>
-            {items.map((item) => {
+          <div role="region" id="itemsList" aria-live="polite">
```

Comment:
I love your train of thought here. Unfortunately, this live region does us no good. 

In order for a live region to work, it must be registered in the DOM _before_ any of the changes it watches for, and it must persist. This entire view is one component – that means that, as you filter, React is re-rendering _everything_, including your live region. All of this re-rendering means that no changes are ever sent to assistive technologies.  :(

There is no way you could have known this, so it's totally fine! There are solutions, and I **definitely** want to talk about them as soon as I figure out the best way to share my ideas with the entire team. 

In the meantime, could one of you please make an issue so that we all remember to come back and implement a live region later?

Reply:
Thanks for the heads up! I went ahead and created the issue: #34